utils/services/goodreads_dataset_parser.py

```diff
@@ -0,0 +1,37 @@
+import csv
+import json
+
+ENTRY_INDEX = 0
+TITLE_INDEX = 1
+AUTHOR_INDEX = 2
+ISBN_INDEX = 4
+
+# Iterates over the Goodreads-books dataset, converting each entry
+# into a book dictionary compatible with the existing CoffeeHouse
+# architecture.
+def parse_goodreads_dataset():
+  goodreads_dataset = open("../data/books.csv")
+  goodreads_csv = csv.reader(goodreads_dataset)
+
+  book_list = []
+  for index, row in enumerate(goodreads_csv):
+    # Skip CSV file column headers.
+    if index == 0: continue
+
+    title = row[TITLE_INDEX]
+    author = row[AUTHOR_INDEX]
+    isbn = row[ISBN_INDEX]
+    
+    book = {
+      "title": title,
+      "author": author,
+      "isbn": isbn,
+    }
+    book_json = json.dumps(book)
+    book_list.append(book_json + "\n")
+  
+  booksFile = open("../data/parsed_books.txt", "w")
+  booksFile.writelines(book_list)
+  booksFile.close()
```

Comment:
It seems like you accumulate a __lot__ of stuff in memory, then write it into the file all at once. Why not open this file with a `with` statement at the top of this function and write to it from inside the for loop like `books_file.writelines((book_json, '\n'))`. Then you can skip accumulating the books in that big array.